src/Passes/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+cmake_minimum_required(VERSION 3.4.3)
+
+project(QSharpPasses)
+
+find_package(LLVM REQUIRED CONFIG)
+include(CheckCXXCompilerFlag)
+
+message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
+message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
+
+# Setting the standard configuration for the C++ compiler
+# Rather than allowing C++17, we restrict the ourselves to 
+# C++14 as this is the standard currently used by LLVM. While
+# there is a very small change that the difference in standard 
+# would break things, it is a possibility nonetheless.
```

Comment:
(Just curiosity)
What does it mean: "C++14 .. this is the standard currently used by LLVM"? As I know LLVM deals with IR rather than C++ code (and Clang is the one who deals with C and C++ code). 

Did you mean "C++14 .. is the standard currently used by _Clang_"? If so, then what does that phrase mean again? I thought that [Clang fully supports C++17 since Clang 5](https://clang.llvm.org/cxx_status.html#cxx17), and the stable version of Clang is at least Clang 12 (or maybe even newer).

To summarize I would recommend to consider C++20 and C++17 first, and if there is a strong reason, then C++14. For now I just don't see the reasons for limiting ourselves with C++14.

Reply:
LLVM is written in C++ and by default they use `C++14` when the compile the library. Therefore, it is advisable to use `C++14` to avoid any linking problems that may arise from using different standards. This is unrelated to what kind of compilers are currently implemented by LLVM. I guess a more accurate formulation would be `C++14 as this is the standard currently used by LLVM project for compilation.`. I've updated the text - let me know if it is still unclear.

Reply:
Aha... The passes are compiled to dynamic libraries that are dynamically loaded by an executable written in C++14. That is why it is desirable for the dynamic libraries to be also written in C++14. Totally makes sense now. Thank you for clarification. Marking this discussion as resolved.